storage/tianmu/handler/tianmu_handler.cpp

```diff
@@ -544,10 +544,7 @@ int TianmuHandler::delete_row([[maybe_unused]] const uchar *buf) {
  Called from sql_select.cc by JOIN::rein*it.
  Called from sql_union.cc by st_select_lex_unit::exec().
  */
-int TianmuHandler::delete_all_rows() {
```

Comment:
Do we do access check(priviledges check)?

Reply:
In the Tianmu engine, there is no need to judge the execution permission. The judgment of permission is already done at the beginning of the execution of the "execute" method in the SQL layer.
for example
#### for example 1:
```c++
bool Sql_cmd_delete::execute(THD *thd)
{
  assert(thd->lex->sql_command == SQLCOM_DELETE);

  LEX *const lex= thd->lex;
  SELECT_LEX *const select_lex= lex->select_lex;
  SELECT_LEX_UNIT *const unit= lex->unit;
  TABLE_LIST *const first_table= select_lex->get_table_list();
  TABLE_LIST *const all_tables= first_table;

  if (delete_precheck(thd, all_tables))
    return true;
.............
```
#### for example 2:
```c++
bool Sql_cmd_delete_multi::execute(THD *thd)
{
  assert(thd->lex->sql_command == SQLCOM_DELETE_MULTI);

  bool res= false;
  LEX *const lex= thd->lex;
  SELECT_LEX *const select_lex= lex->select_lex;
  TABLE_LIST *const first_table= select_lex->get_table_list();
  TABLE_LIST *const all_tables= first_table;

  TABLE_LIST *aux_tables= thd->lex->auxiliary_table_list.first;
  uint del_table_count;
  Query_result_delete *del_result;

  if (multi_delete_precheck(thd, all_tables))
    return true;
........
```

#### for example 3:
```c++
bool Sql_cmd_truncate_table::execute(THD *thd)
{
  bool res= TRUE;
  TABLE_LIST *first_table= thd->lex->select_lex->table_list.first;
  DBUG_ENTER("Sql_cmd_truncate_table::execute");

  if (check_one_table_access(thd, DROP_ACL, first_table))
    DBUG_RETURN(res);
.......
```

> They all call check_ access() and check_ grant() judge the execution permission